src/element/mod.rs

```diff
@@ -453,6 +466,13 @@ impl Element {
         &self.value
     }
 
+    /// Returns the source location metadata (row, column).
+    /// If element was constructed from a `LazyValue` then it returns `Some(_)`.
+    /// Otherwise, returns `None` for programmatically constructed element.
+    pub fn location(&self) -> Option<(usize, usize)> {
```

Comment:
I don't want to blow up the scope of this PR, so can you create an issue to re-evaluate the public API for this before we release this feature? I still think that we should consider having a (possibly opaque) `SourceLocation` type.

Reply:
#948